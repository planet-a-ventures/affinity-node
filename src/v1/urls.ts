/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#get-a-specific-list) for more info.
 */
export const listsUrl = (list_id?: number) =>
    list_id ? `/lists/${encodeURIComponent(list_id)}` : '/lists'

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#get-a-specific-list-entry) for more info.
 */
export const listEntriesUrl = (list_id: number, list_entry_id?: number) =>
    listsUrl(list_id) + (
        list_entry_id
            ? `/list-entries/${encodeURIComponent(list_entry_id)}`
            : '/list-entries'
    )

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#rate-limit) for more info.
 */
export const rateLimitUrl = () => '/rate-limit'

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#whoami) for more info.
 */
export const whoAmIUrl = () => '/auth/whoami'

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#fields) for more info.
 */
export const fieldsUrl = (field_id?: number) =>
    field_id ? `/fields/${encodeURIComponent(field_id)}` : '/fields'

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#field-values) for more info.
 */
export const fieldValuesUrl = (field_value_id?: number) => {
    return field_value_id
        ? `/field-values/${encodeURIComponent(field_value_id)}`
        : '/field-values'
}

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#get-field-values-changes) for more info.
 */
export const fieldValueChangesUrl = () => '/field-value-changes'

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#organizations) for more info.
 */
export const organizationsUrl = (organization_id?: number | 'fields') => {
    return organization_id
        ? `/organizations/${encodeURIComponent(organization_id)}`
        : '/organizations'
}

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#get-global-organizations-fields) for more info.
 */
export const organizationFieldsUrl = () => organizationsUrl('fields')

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#persons) for more info.
 */
export const personsUrl = (person_id?: number | 'fields') => {
    return person_id ? `/persons/${encodeURIComponent(person_id)}` : '/persons'
}

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#get-global-person-fields) for more info.
 */
export const personFieldsUrl = () => personsUrl('fields')

/**
 * @hidden
 * See [here](https://api-docs.affinity.co/#entity-files) for more info.
 */
export const entityFilesUrl = (
    entity_file_id?: number,
    is_download: boolean = false,
) => {
    return entity_file_id
        ? `/entity-files` + (is_download ? '/download' : '') +
            `/${encodeURIComponent(entity_file_id)}`
        : '/entity-files'
}
