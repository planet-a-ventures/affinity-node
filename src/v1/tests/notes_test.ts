import { afterEach, beforeEach, describe, it } from '@std/testing/bdd'
import { assertSnapshot } from '@std/testing/snapshot'
import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'
import { Affinity } from '../index.ts'
import type { AllNotesRequest } from '../notes.ts'
import { notesUrl } from '../urls.ts'
import { apiKey, isLiveRun } from './env.ts'
import { getRawFixture } from './get_raw_fixture.ts'
import { mockPagingFromAllEndpoint } from './mock_paging_from_all_endpoint.ts'

describe('notes', () => {
    let mock: MockAdapter
    let affinity: Affinity

    beforeEach(() => {
        if (!isLiveRun()) {
            mock = new MockAdapter(axios, { onNoMatch: 'throwException' })
        }
        affinity = new Affinity(apiKey() || 'api_key')
    })
    afterEach(() => {
        mock?.reset()
    })

    it('can list all notes', async (t) => {
        mock?.onGet(notesUrl()).reply(
            200,
            await getRawFixture('notes/all.raw.response.json'),
        )
        const res = await affinity.notes.all({
            organization_id: 297551634,
        })
        await assertSnapshot(t, res)
    })

    it('iterates over all notes', async (t) => {
        const params: AllNotesRequest = {
            organization_id: 297551634,
            page_size: 1,
        }

        await mockPagingFromAllEndpoint(
            './fixtures/notes/all.raw.response.json',
            params,
            'notes',
            notesUrl,
            mock,
        )

        let page = 0
        for await (
            const entries of affinity.notes.pagedIterator(params)
        ) {
            await assertSnapshot(t, entries, {
                name: `page ${++page} of notes`,
            })
        }
    })
})
