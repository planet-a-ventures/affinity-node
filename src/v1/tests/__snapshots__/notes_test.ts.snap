export const snapshot = {};

snapshot[`notes > can list all notes 1`] = `
{
  next_page_token: null,
  notes: [
    {
      associated_person_ids: [],
      content: "Test ***html***

* 1

* 2

* 3",
      created_at: 2024-08-13T10:23:07.846Z,
      creator_id: 54576635,
      id: 20421456,
      interaction_id: null,
      interaction_person_ids: [],
      interaction_type: null,
      is_meeting: false,
      mentioned_person_ids: [],
      opportunity_ids: [],
      organization_ids: [
        297551634,
      ],
      parent_id: null,
      person_ids: [],
      type: 2,
      updated_at: null,
    },
    {
      associated_person_ids: [],
      content: "Test text",
      created_at: 2024-08-13T10:22:45.964Z,
      creator_id: 54576635,
      id: 20421454,
      interaction_id: null,
      interaction_person_ids: [],
      interaction_type: null,
      is_meeting: false,
      mentioned_person_ids: [],
      opportunity_ids: [],
      organization_ids: [
        297551634,
      ],
      parent_id: null,
      person_ids: [],
      type: 2,
      updated_at: null,
    },
  ],
}
`;

snapshot[`page 1 of notes 1`] = `
[
  {
    associated_person_ids: [],
    content: "Test ***html***

* 1

* 2

* 3",
    created_at: 2024-08-13T10:23:07.846Z,
    creator_id: 54576635,
    id: 20421456,
    interaction_id: null,
    interaction_person_ids: [],
    interaction_type: null,
    is_meeting: false,
    mentioned_person_ids: [],
    opportunity_ids: [],
    organization_ids: [
      297551634,
    ],
    parent_id: null,
    person_ids: [],
    type: 2,
    updated_at: null,
  },
]
`;

snapshot[`page 2 of notes 1`] = `
[
  {
    associated_person_ids: [],
    content: "Test text",
    created_at: 2024-08-13T10:22:45.964Z,
    creator_id: 54576635,
    id: 20421454,
    interaction_id: null,
    interaction_person_ids: [],
    interaction_type: null,
    is_meeting: false,
    mentioned_person_ids: [],
    opportunity_ids: [],
    organization_ids: [
      297551634,
    ],
    parent_id: null,
    person_ids: [],
    type: 2,
    updated_at: null,
  },
]
`;

snapshot[`page 3 of notes 1`] = `[]`;
